.github/workflows/gosec-scan.yaml

```diff
@@ -0,0 +1,57 @@
+# Copyright 2024 The Nephio Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: Gosec security scan
+
+on:
+  push:
+    paths-ignore:
+      - "docs/**"
+      - "release/**"
+      - ".prow.yaml"
+      - "OWNERS"
+  pull_request:
+    paths-ignore:
+      - "docs/**"
+      - "release/**"
+      - ".prow.yaml"
+      - "OWNERS"
+
+jobs:
+  tests:
+    name: Porch gosec scan
+    runs-on: ubuntu-latest
+    permissions:
+      # required for all workflows
+      security-events: write
+    env:
+      GO111MODULE: on
+    steps:
+      - name: Checkout Porch
+        uses: actions/checkout@v4
+      - name: Set up Go
+        uses: actions/setup-go@v5
+        with:
+          go-version: '>=1.22.2'
+      - name: Run Gosec Security Scanner
```

Comment:
suggestion: Would it be a good idea to add a step to invoke the target in the makefile as well such as below to ensure consistency?
```suggestion
      - name: Run Gosec Security Scanner using the makefile target
        run:
           make gosec GOSEC_VER=v2.21.4
```